### PR TITLE
`r_base::sum()` handling `na.rm = `

### LINKED ISSUE #70: Extension `sum()` function
Follow up on https://github.com/duckdblabs/duckplyr/issues/130#issuecomment-2022736434

### PR DESCRIPTION
**Pull request by romainfrancois**
closes #70

### CODE REVIEW

File: src/sum.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#include "duckdb/core_functions/aggregate/sum_helpers.hpp"
+#include "duckdb/core_functions/aggregate/distributive_functions.hpp"
+
+namespace duckdb {
+namespace rfuns {
+
+unique_ptr<FunctionData> BindRSum(ClientContext &context, AggregateFunction &function, vector<unique_ptr<Expression>> &arguments) {
+	function = SumFun::GetFunctions().GetFunctionByArguments(context, {arguments[0]->return_type});
+	return nullptr;
+}
```
Comment (romainfrancois) [inline src/sum.cpp:16]: TODO: handle `arguments[1]` here, i.e. if it is `TRUE` go through the data and if we find a null, then bind to a function that returns a null somehow.

Comment (krlmlr) [reply src/sum.cpp:16]: You can assume a constant here, duckplyr won't let anything else through.

**Review by krlmlr — APPROVED** [review]
Nice, thanks!

**Comment by romainfrancois** [comment]
handling `na.rm = FALSE`: 

``` r
library(duckdbrfuns)

rfuns_sum(1:10)
#> [1] 55
rfuns_sum(c(1:10, NA))
#> [1] 55

rfuns_sum(c(1:10, NA), na.rm = TRUE)
#> [1] 55
rfuns_sum(c(1:10, NA), na.rm = FALSE)
#> [1] NA

rfuns_sum(integer(), na.rm = FALSE)
#> [1] 0
rfuns_sum(integer(), na.rm = TRUE)
#> [1] 0

rfuns_sum(NA_integer_, na.rm = TRUE)
#> [1] 0
rfuns_sum(NA_integer_, na.rm = FALSE)
#> [1] NA
```

<sup>Created on 2024-04-24 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Review by krlmlr — APPROVED** [review]
Thanks. We also want `sum(logical())`, but this could be a separate PR.

**Pull request merged**